idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/editor/lineMarker/LineMarkerSyntax.kt

```diff
@@ -39,32 +43,113 @@ interface LineMarkerSyntax {
     icon: Icon,
     transform: (PsiElement) -> A?,
     message: (element: A) -> String = Noop.string1(),
-    placed: GutterIconRenderer.Alignment = GutterIconRenderer.Alignment.RIGHT
+    placed: GutterIconRenderer.Alignment = GutterIconRenderer.Alignment.RIGHT,
+    navigate: (event: MouseEvent, element: PsiElement) -> Unit = Noop.effect2,
```

Comment:
from a user's perspective: 
- how should `navigate` be used? Perhaps add a `@param`? What will be value of `element`, is this the return value of `transform`? 
- is `clickAction` similar? Could both be combined? Perhaps add a `@param`?

Reply:
Done :)